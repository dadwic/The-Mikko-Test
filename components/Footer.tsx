import React from "react";
import Box from "@mui/material/Box";
import Link from "@mui/material/Link";
import Typography from "@mui/material/Typography";
import Container from "@mui/material/Container";

function Copyright() {
  return (
    <Typography variant="body2" color="text.secondary">
      {`Copyright © ${new Date().getFullYear()}`}
    </Typography>
  );
}

export default function Footer() {
  return (
    <Box
      component="footer"
      sx={{
        py: 2,
        px: 2,
        mt: "auto",
      }}
    >
      <Container maxWidth="sm">
        <Typography variant="body1">
          Made with ❤️ by{" "}
          <Link
            color="inherit"
            target="_blank"
            href="https://github.com/dadwic"
          >
            @dadwic
          </Link>
        </Typography>
        <Copyright />
      </Container>
    </Box>
  );
}
