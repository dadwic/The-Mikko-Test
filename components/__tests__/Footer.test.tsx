import React from "react";
import renderer from "react-test-renderer";
import Footer from "../Footer";

describe("Footer", () => {
  it("should correctly match snapshot", () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
