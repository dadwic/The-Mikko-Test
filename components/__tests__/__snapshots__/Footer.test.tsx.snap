// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should correctly match snapshot 1`] = `
<footer
  className="MuiBox-root css-1pjb125"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Made with ❤️ by
       
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
        href="https://github.com/dadwic"
        onBlur={[Function]}
        onFocus={[Function]}
        target="_blank"
      >
        @dadwic
      </a>
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
    >
      Copyright © 2022
    </p>
  </div>
</footer>
`;

exports[`renders unchanged 1`] = `
<div>
  <footer
    class="MuiBox-root css-1pjb125"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Made with ❤️ by
         
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
          href="https://github.com/dadwic"
          target="_blank"
        >
          @dadwic
        </a>
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        Copyright © 2022
      </p>
    </div>
  </footer>
</div>
`;
